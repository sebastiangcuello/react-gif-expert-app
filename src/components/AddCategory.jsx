import { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = ( { target } ) => {
        setInputValue(target.value);
    }

    const onSubmit = ( event ) => {
        event.preventDefault();       
        if( inputValue.trim().length <= 1 ) return;     

        // setCategories( categories => [ inputValue, ...categories] );
        setInputValue('');
        onNewCategory( inputValue.trim() );
    }

  return (
    //NO USO FRAGMENTO EN ESTE CASO PORQUE ES 1 SOLO ELEMENTO, Y EL RESTO SON HIJOS
    <form onSubmit={ onSubmit } aria-label="form">
        <input
            type="text"
            placeholder="Buscar gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
    
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}