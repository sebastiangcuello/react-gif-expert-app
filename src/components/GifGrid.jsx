//https://reactjs.org/docs/strict-mode.html

import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    //NO COLOCAR FUNCIONES DENTRO DE UN FUNCTIONAL COMPONENT, PORQUE SE EJECUTARÍA CADA VEZ QUE CAMBIA DE ESTADO...

    //El primer parametro es el callback, es decir la función que quiero ejecutar
    //El 2do parametro es una lista de las dependencias que se tienen que cumplir para que se ejecute el callback
    //Sí dejo el array de dependencias VACIO [], entonces se ejecuta UNA SOLA VEZ el callback.

    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
        
            <h3>{ category }</h3>
            {/* <LoadingMessage isLoading = { isLoading }/> */}
            {
              isLoading && ( <h2> Cargando... </h2> )
            }
            
            <div className="card-grid">
            {     
                images.map( ( image ) => (
                     <GifItem 
                        key= { image.id }
                        { ...image }   //para envíar todas las properties
                     />
                ))      
            }
            </div>

        </>
    )
}
