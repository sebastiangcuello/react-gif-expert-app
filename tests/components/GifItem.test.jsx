import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas de <GifItem />', () => { 
    
    const title = 'Goku';    
    const url = 'https://one-punch.com/Goku';    

    test('should de hacer match con el snapshot', () => { 
        
        const { container } = render( <GifItem title = { title } url = { url } /> );
        expect( container ).toMatchSnapshot();

    })

    test('debe de mostrar la imagen con el URl y el ALT indicado', () => { 
        
        render( <GifItem title = { title } url = { url } />); 
        //screen.debug();

        // Podemos testear campo a campo
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        //O podemos testear desestructurar

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
        
     });

     test('debe de mostrar el titulo en el componente', () => { 
        
        render( <GifItem title = { title } url = { url }/>);
        expect( screen.getByText( title )).toBeTruthy();

      });

 });