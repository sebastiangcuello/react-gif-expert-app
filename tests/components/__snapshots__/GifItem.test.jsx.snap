// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem /> should de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://one-punch.com/Goku"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
